datafusion/physical-expr/src/expressions/literal.rs

```diff
@@ -74,8 +74,13 @@ impl PhysicalExpr for Literal {
 }
 
 /// Create a literal expression
-pub fn lit(value: ScalarValue) -> Arc<dyn PhysicalExpr> {
-    Arc::new(Literal::new(value))
+pub fn lit<T: datafusion_expr::Literal>(value: T) -> Arc<dyn PhysicalExpr> {
```

Comment:
Here is the actual change. It isn't the most beautiful code but I think it is reasonable

Reply:
This could also be written slightly more concisely as a match:

```
    let scalar_value = match value.lit() {
      Expr::Literal(v) => v,
      _ => unreachable!()
    };
```

Reply:
I implemented this suggestion in https://github.com/apache/arrow-datafusion/pull/2838